week6/Filmer/app/src/main/java/com/tahhan/filmer/database/MovieDao.kt

```diff
@@ -7,21 +7,22 @@ import androidx.room.Insert
 import androidx.room.OnConflictStrategy.IGNORE
 import androidx.room.Query
 import com.tahhan.filmer.model.Movie
+import kotlinx.coroutines.flow.Flow
 
 
 @Dao
 abstract class MovieDao {
 
     @Insert(onConflict = IGNORE)
-    abstract fun insertMovie(movie: Movie?)
+    abstract suspend fun insertMovie(movie: Movie?)
 
     @Delete
-    abstract fun deleteMovie(movie: Movie?)
+    abstract suspend  fun deleteMovie(movie: Movie?)
 
 
     @Query("Select * from movies order by popularity Asc")
-    abstract fun loadFavoriteMovies(): LiveData<List<Movie>>
+    abstract fun loadFavoriteMovies(): Flow<List<Movie>>
 
     @Query("SELECT * FROM movies WHERE id = :id")
-    abstract fun loadMovieById(id: String?): Movie
+    abstract  fun loadMovieById(id: String?): Flow<Movie>
```

Comment:
I would personally refrain from using observable structures for single items, unless you have specific "Update" options and features in your app, that would render some changes and emit them later on